mlflow/xgboost.py

```diff
@@ -130,7 +138,29 @@ def save_model(
                           model. The given example will be converted to a Pandas DataFrame and then
                           serialized to json using the Pandas split-oriented format. Bytes are
                           base64-encoded.
-
+    :param pip_requirements: Overwrites a default set of pip requirements generated automatically
+        by the model flavor on the user's current software environment. The given requirements are
+        automatically parsed and logged as a single `requirements.txt` file. Either string path to
+        a pip requirements file on the local filesystem (e.g. ``"requirements.txt"``) or an iterable
+        of pip requirement strings (e.g. ``["scikit-learn", "-r requirements.txt"]``) is acceptable.
+    :param additional_pip_requirements: Specifies additional pip requirements on top of a default
+        set of pip requirements generated automatically by the model flavor on the user's current
+        software environment. The acceptable value type is the same as ``pip_requirements``.
+
+        .. note::
+            It is recommended to use ``pip_requirements`` and ``additional_pip_requirements``
+            when specifying pip requirements rather than ``conda_env`` which has been deprecated
+            and only exists for backwards compatibility.
```

Comment:
@dbczumar I've addressed your comment on these parameters. What do you think? If this looks good, I'll address https://github.com/mlflow/mlflow/pull/4519/files#r665882073